### PR TITLE
Gallery finished

### PR DESCRIPTION
**Pull request by Samuel-Fox**
# Description

Added a gallery screen that lets the users rotate through the different pieces in the game along with a short description and artist credits to accompany each piece.

## Related issue(s)

#11 

## List of changes

A more detailed list of the changes.
- Added the Gallery as a new scene
- Gallery button from main screen switches to the gallery when clicked
- Next and Previous buttons loop users through an array on pieces and a description of the piece

## Additional notes

Open to changing UI if some other layout of the elements is preferable. Do a double check on grammar of descriptions and make sure I did not miscredit any artists, I made a list to make sure each artist got the proper credit but it's not impossible I missed something.

### CODE REVIEW

File: Birthday-2024-Project/Scripts/UI/MainMenu/MainMenuUiManager.gd [hunk 1]
```diff
@@ -41,11 +42,15 @@ func show_campaign_levels():
 	_disable_all_screans()
 	campaign_levels.show()
 
+func show_gallery_screen():
+	_disable_all_screans()
+	gallery_screen.show()
 
 func _disable_all_screans():
```
Comment (Mikotey) [inline Birthday-2024-Project/Scripts/UI/MainMenu/MainMenuUiManager.gd:49]: I never noticed this said "screans" lol

**Review by Mikotey — COMMENTED** [review]
Screen looks good! I like the placements of the pieces and text. All the buttons are setup and working properly.

@Casulono We might need to review the piece descriptions. They seem to be missing articles (a, an, the) in parts. And it feels like some have run on sentences.

This PR might have to wait until the descriptions get updated.
Or, since all the logic is in place, we could push the PR and update the text later with another PR. This PR uses the older style of main menu screen management. Thus, I'll probably need to merge it into my PR and resolve the conflicts

image
There are these arrow assets (under Art/UI) that I started using for the level select screens. I think it would look nice if we swapped the previous and next buttons with these arrows. What do you think?
You can checkout the implementation in MenuRefactorStates PR branch if you want to interact with the buttons

**Comment by Casulono** [comment]
@Samuel-Fox @Mikotey Went through several online checkers and also did a manual one. Here is the result: https://docs.google.com/document/d/1qsYnMhI8IoCFPuJJNmh6ylV0C0Hmzwu9wFTi8Myj2A0/edit

File: Birthday-2024-Project/ScenePrefabs/gallery.tscn [hunk 2]
```diff
@@ -0,0 +1,370 @@
+[gd_scene load_steps=31 format=3 uid="uid://k787vo3otomq"]
```
Comment (Casulono) [inline Birthday-2024-Project/ScenePrefabs/gallery.tscn:1]: minor thing: change the name of the scene to "Gallery" to have it consistent with other scenes

**Review by Casulono — CHANGES_REQUESTED** [review]

**Comment by Samuel-Fox** [comment]
Pushed new changes

- Added name of the pieces above the descriptions
- Update descriptions from the google doc
- Changed UI to more closely match the one from the level selection
- Fixed capitalization on gallery

File: Birthday-2024-Project/ScenePrefabs/MainMenuUiManager.tscn [hunk 3]
```diff
@@ -1,12 +1,13 @@
-[gd_scene load_steps=6 format=3 uid="uid://gfva2owqvlhw"]
+[gd_scene load_steps=7 format=3 uid="uid://gfva2owqvlhw"]
 
 [ext_resource type="Script" path="res://Scripts/UI/MainMenu/MainMenuUiManager.gd" id="1_lofx7"]
 [ext_resource type="Script" path="res://Scripts/UI/MainMenu/MainMenuStartScreen.gd" id="2_n7aum"]
 [ext_resource type="PackedScene" uid="uid://4kktp1gwthbt" path="res://ScenePrefabs/CampaignSelectMenu.tscn" id="3_qksbd"]
 [ext_resource type="Script" path="res://Scripts/UI/MainMenu/States/MainMenuUiStartState.gd" id="3_vnjds"]
 [ext_resource type="PackedScene" uid="uid://lma3uuskmlpo" path="res://ScenePrefabs/CampaignLevelSelectMenu.tscn" id="4_o3atr"]
+[ext_resource type="PackedScene" uid="uid://k787vo3otomq" path="res://ScenePrefabs/Gallery.tscn" id="5_1folx"]
```
Comment (Casulono) [inline Birthday-2024-Project/ScenePrefabs/MainMenuUiManager.tscn:8]: actual gallery.tscn file is still without a capital letter

File: Birthday-2024-Project/ScenePrefabs/gallery.tscn [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
...oses. And you, knowing all this, are staring at him! Give him some privacy!", "Where did Fauna go? She could be anywhere. She could be in her closet. She could be inside your walls. She could be right behind you! Or she could be somewhere here in the bushes.", "Did you know that green and purple colors are a great combination and are very pleasant for the eyes? Here we have Exhibit A: BEEGsmol Faunya in a purple dress, accompanied by the green of a spilled matcha latte.", "Even more matcha? What will I do with this whole powder? One glass is not enough. On the other hand, is it possible to overdose on it?", "Cowkini? Check. Safety cap? Check. Matcha latte? Che… Aaaaa… Uuuuu… That was not a good idea. Oh well…", "If, by any chance, your enemies have medieval castles, just follow EleFauna’s advice. “Prrrt. Prrrt. Toot~ Elephants, roll out!” It works every time.", "“Okaerinasaimase, Goshujin-sama! Your... your... your maid is here. It was me all along. Aaaaaaa! What do you think? What do you think?”", "\"Konfauna~ Your gaming idol kirin Ceres Fauna is here!\" The Keeper of Nature, a druidic kirin who appeared on the internet to win over humans and convince them to return to nature."])
```
Comment (Casulono) [inline Birthday-2024-Project/ScenePrefabs/gallery.tscn:59]: "she will provide commentary for the participants in the GoI Trials! for the participants of GOI Trials!" remove one instance of "for the participants of GOI Trials!"

Comment (Casulono) [reply Birthday-2024-Project/ScenePrefabs/gallery.tscn:59]: I've updated in doc "is it possible to overdose on it?" to "is it possible to overdose it?". Should also be pulled here

**Review by Casulono — CHANGES_REQUESTED** [review]

File: Birthday-2024-Project/ScenePrefabs/MainMenuUiManager.tscn (hunk 3 above)
Comment (Samuel-Fox) [reply Birthday-2024-Project/ScenePrefabs/MainMenuUiManager.tscn:8]: Should be fixed now

File: Birthday-2024-Project/ScenePrefabs/gallery.tscn (hunk 4 above)
Comment (Samuel-Fox) [reply Birthday-2024-Project/ScenePrefabs/gallery.tscn]: Fixed both instances of incorrect descriptions

**Review by Casulono — APPROVED** [review]

**Pull request merged**